elyra/pipeline/validate.py

```diff
@@ -0,0 +1,552 @@
+                                                   "nodeName": node['app_data']['ui_data']['label'],
+                                                   "pipelineID": pipeline_id})
+        else:
+            response.add_message(severity=ValidationSeverity.Error,
+                                 message_type="invalidRuntime",
+                                 message="Unsupported pipeline runtime selected in this pipeline",
+                                 data={"pipelineRuntime": pipeline_runtime,
+                                       "pipelineID": pipeline_id})
+
+    async def _validate_node_properties(self, root_dir: str, pipeline: Dict, pipeline_runtime: str,
+                                        pipeline_execution: str, response: ValidationResponse) -> None:
+        """
+        Validates each of the node's structure for required fields/properties as well as
+        their values
+        :param root_dir: the absolute base path of the current elyra workspace
+        :param pipeline: the pipeline definition to be validated
+        :param pipeline_runtime: name of the pipeline runtime being used e.g. kfp, airflow, generic
+        :param response: ValidationResponse containing the issue list to be updated
+        """
+        pipeline_json = json.loads(json.dumps(pipeline))
+
+        for single_pipeline in pipeline_json['pipelines']:
+            node_list = single_pipeline['nodes']
+            pipeline_runtime = 'local' if pipeline_runtime == 'generic' else pipeline_runtime
+            component_list = await PipelineProcessorManager.instance().get_components(pipeline_runtime)
+            components = json.loads(ComponentRegistry.to_canvas_palette(component_list))
+            for node in node_list:
+                if node['type'] == 'execution_node':
+                    node_data = node['app_data']
+                    if node['op'] in ['execute-r-node', 'execute-python-node', 'execute-notebook-node']:
+                        # Validate actual node property values
+                        self._validate_filepath(node=node, root_dir=root_dir, property_name='filename',
+                                                filename=node_data['filename'], response=response)
+
+                        # If the running locally, we can skip the resource and image name checks
+                        if pipeline_execution != 'local':
+                            self._validate_container_image_name(node, response=response)
+                            for resource_name in ['cpu', 'gpu', 'memory']:
+                                if resource_name in node_data.keys():
+                                    self._validate_resource_value(node, resource_name=resource_name, response=response)
+                        if pipeline_runtime == 'kfp' and node_data['filename'] != node_data['ui_data']['label']:
+                            self._validate_ui_data_label(node=node, label_name=node_data['ui_data']['label'],
```

Comment:
Heads up.  #1882 moves the label field to app_data.

Reply:
Rather than pass `node` to every validation method, it might be cleaner to extract the node-id and node-name information once and pass them individually and `node` only when necessary.

I wonder if it would make since to have a NodeValdation class that takes the node and response in its constructor and has a simple `validate()` method where each of these definitions reside?  This would enable use of instance variables and reduce parameters, etc.  Not a big deal though.